wordpress-mobile/WordPress-Android
# Prepublishing bottomsheet categories not loading in story creator

Fixes #14234

## Findings
`TaxonomyActionBuilder.newFetchCategoriesAction(site)` was not being used in the `PrepublishingViewModel` to do an initial fetch of categories, so if the user did not attempt to create a post and modify the categories via Post Settings, the local cache will be empty.

Another issue that was mentioned was 

>  when visiting the categories screen from My Site > Settings > Default Category, that is also blank.

I was not able to reproduce this. I will give it another try tomorrow.

## Solution

1. Use `TaxonomyActionBuilder.newFetchCategoriesAction(site)` to fetch the categories when the bottom sheet is loaded so that when you navigate to the Categories you will see them. 

2. I did a cleanup of some unused functions & params.

## Testing
From a fresh install of the app:

1. Enter the story creator
2. Press 'publish'
3. Tap the categories option
4. Notice that categories load

## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist
